File: app/backend/java/payment-service/src/main/java/com/srbms/paymentservice/dto/PaymentRequestDTO.java
package com.srbms.paymentservice.dto;

import com.srbms.paymentservice.entity.PaymentMode;
import jakarta.validation.constraints.DecimalMin;
import jakarta.validation.constraints.NotNull;
import java.math.BigDecimal;

public class PaymentRequestDTO {

    @NotNull(message = "Ride ID is required")
    private Integer rideId;

    @NotNull(message = "User ID is required")
    private Integer userId;

    @NotNull(message = "Total fare is required")
    @DecimalMin(value = "0.01", message = "Total fare must be greater than zero")
    private BigDecimal totalFare;

    private BigDecimal discountAmount;

    @NotNull(message = "Payment mode is required")
    private PaymentMode paymentMode;

    private String transactionId;
    private String paymentDetailsToken; // Encrypted card/UPI details simulation

    public PaymentRequestDTO() {}

    public Integer getRideId() {
        return rideId;
    }

    public void setRideId(Integer rideId) {
        this.rideId = rideId;
    }

    public Integer getUserId() {
        return userId;
    }

    public void setUserId(Integer userId) {
        this.userId = userId;
    }

    public BigDecimal getTotalFare() {
        return totalFare;
    }

    public void setTotalFare(BigDecimal totalFare) {
        this.totalFare = totalFare;
    }

    public BigDecimal getDiscountAmount() {
        return discountAmount;
    }

    public void setDiscountAmount(BigDecimal discountAmount) {
        this.discountAmount = discountAmount;
    }

    public PaymentMode getPaymentMode() {
        return paymentMode;
    }

    public void setPaymentMode(PaymentMode paymentMode) {
        this.paymentMode = paymentMode;
    }

    public String getTransactionId() {
        return transactionId;
    }

    public void setTransactionId(String transactionId) {
        this.transactionId = transactionId;
    }

    public String getPaymentDetailsToken() {
        return paymentDetailsToken;
    }

    public void setPaymentDetailsToken(String paymentDetailsToken) {
        this.paymentDetailsToken = paymentDetailsToken;
    }
}

File: app/backend/.net/SmartRide.PaymentService/Models/DTOs/PaymentRequestDTO.cs
using System.ComponentModel.DataAnnotations;
using SmartRide.PaymentService.Models.Enums;

namespace SmartRide.PaymentService.Models.DTOs;

public class PaymentRequestDTO
{
    [Required(ErrorMessage = "Ride ID is required")]
    public int RideId { get; set; }

    [Required(ErrorMessage = "User ID is required")]
    public int UserId { get; set; }

    [Required(ErrorMessage = "Total fare is required")]
    [Range(typeof(decimal), "0.01", "79228162514264337593543950335", ErrorMessage = "Total fare must be greater than zero")]
    public decimal TotalFare { get; set; }

    public decimal? DiscountAmount { get; set; }

    [Required(ErrorMessage = "Payment mode is required")]
    public PaymentMode PaymentMode { get; set; }

    public string? TransactionId { get; set; }
    public string? PaymentDetailsToken { get; set; }
}
